dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/DataOkHttpUploader.kt

```diff
@@ -15,19 +15,21 @@ import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody
 
-internal class DataOkHttpUploader(
+internal abstract class DataOkHttpUploader(
```

Comment:
I'm wondering if the `abstract` class is indeed needed. The only variable dependency seems to be `uploadUrlFormat ` and the rest of this class uses just `DataOkHttpUploader's` logic. Have you considered using just two instances of `DataOkHttpUploader`?
```swift
let logsUploader = DataOkHttpUploader(...)
let tracesUploader = DataOkHttpUploader(...)
```

In iOS SDK I introduced `LogsUploadStrategy` which is the abstraction building log upload handler. For tracing I might then have `LogsUploadStrategy` and `TracesUploadStrategy`. WDYT?

Reply:
I think for now I will keep it simple. The goal is to have the TracesUploader and to make it work in this PR. Once we have this first version of tracing ready we can add more abstractions like we did with the Logs.

Reply:
Regarding your question, the answer is yes, this will be in my next commit, I will add the tracesUploader but this will be a separated class (TrackesOkHttpUploader:DataOkHttpUploader). Thing is I need this cause things are pretty uncertain now on the Traces Server API and I want to be as flexible as I can in case something changes.

Reply:
Sounds good 👍